app/main.py

```diff
@@ -1,52 +1,126 @@
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+
+class DisplayStrategy(ABC):
+    @abstractmethod
+    def display(self, content: str) -> None:
+        pass
+
+
+class ConsoleDisplayStrategy(DisplayStrategy):
+    def display(self, content: str) -> None:
+        print(content)
+
+
+class ReverseDisplayStrategy(DisplayStrategy):
+    def display(self, content: str) -> None:
+        print(content[::-1])
+
+
+class PrintStrategy(ABC):
+    @abstractmethod
+    def print(self, title: str, content: str) -> None:
+        pass
+
+
+class ConsolePrintStrategy(PrintStrategy):
+    def print(self, title: str, content: str) -> None:
+        print(f"Printing the book: {title}...")
+        print(content)
+
+
+class ReversePrintStrategy(PrintStrategy):
+    def print(self, title: str, content: str) -> None:
+        print(f"Printing the book in reverse: {title}...")
+        print(content[::-1])
+
+
+class SerializeStrategy(ABC):
+    @abstractmethod
+    def serialize(self, title: str, content: str) -> str:
+        pass
+
+
+class JSONSerializeStrategy(SerializeStrategy):
+    def serialize(self, title: str, content: str) -> str:
+        return json.dumps({"title": title, "content": content})
+
+
+class XMLSerializeStrategy(SerializeStrategy):
+    def serialize(self, title: str, content: str) -> str:
+        root = ElementTree.Element("book")
+        title_elem = ElementTree.SubElement(root, "title")
+        title_elem.text = title
+        content_elem = ElementTree.SubElement(root, "content")
+        content_elem.text = content
+        return ElementTree.tostring(root, encoding="unicode")
+
+
+class BookService:
+    def __init__(
+        self,
+        book: Book,
+        display_strategy: DisplayStrategy,
+        print_strategy: PrintStrategy,
+        serialize_strategy: SerializeStrategy,
```

Comment:
these fields are nullable from what I see
I'd be better to create some dictionary/dictionaries with mapping instead of this class